### PR TITLE
Added basic public-key cryptography example.

### PR DESCRIPTION
**Pull request by DrPeterVanNostrand**

### CODE REVIEW

File: examples/README.md [hunk 1]
```diff
@@ -1,5 +1,11 @@
 # Examples
 
+- [`Public-Key Cryptography`](basic_pkc.rs) - Demonstrates how to: generate a
```
Comment (vkomenda) [inline examples/README.md]: The sentence would sound better without the colon.

File: examples/basic_pkc.rs [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+    // Bob wants to send Alice a message. He signs the plaintext message with his secret-key. He
+    // then encrypts the signed message with Alice's public-key.
+    let msg = b"let's get pizza";
+    let signed_msg = bob.create_signed_msg(msg);
+    let serialized = serialize(&signed_msg).expect("Failed to serialize `SignedMsg`");
+    let ciphertext = alice.pk.encrypt(&serialized);
+
+    // Alice receives Bob's encrypted message. She decrypts the message using her secret-key. She
+    // then verify's that the signature of the plaintext is valid using Bob's public-key.
```
[the remark this answers is not in the document]
Comment (DrPeterVanNostrand) [reply examples/basic_pkc.rs]: > verify's

ugg

**Comment by DrPeterVanNostrand** [comment]
I've heard that you shouldn't use the same private key for signing and decryption, but never really understood why that is. I thought that it was because you may want a "long term" key for proof of identity (i.e. signing), but have "short term" session-based encryption keys (e.g. a single TLS connection). I think that I'm just going to note that this is against best practices, but I can change the code if anyone wants me to.

**Comment by afck** [comment]
Yes, I think just a comment is better.

Apart from the reasons you already mentioned, I think some cryptosystems also allow attacks where I'd give you something to "decrypt" and then can use your decryption share as a _signature share_ for something else. But I don't think that's possible in our case.

**Comment by DrPeterVanNostrand** [comment]
Oh, ok. I see what you mean now. I added a `NOTE:` to the example, I'm going to merge.

**Pull request merged**